app/server/utils/transfer_limits.py

```diff
@@ -157,160 +151,577 @@ def is_any_token_and_user_is_kyc_verified(credit_transfer):
 def is_any_token_and_user_is_kyc_business_verified(credit_transfer):
     return is_user_and_any_token(credit_transfer) and user_business_or_multidoc_kyc_verified(credit_transfer)
 
-# ~~~~~~LIMIT FILTERS~~~~~~
 
+# ~~~~~~LIMIT FILTERS~~~~~~
 
-@curry
-def after_time_period_filter(days: int, query: Query):
 
+def after_time_period_filter(days: int):
     epoch = datetime.datetime.today() - datetime.timedelta(days=days)
+    return [CreditTransfer.created >= epoch]
+
+
+def matching_sender_user_filter(transfer: CreditTransfer):
+    return [CreditTransfer.sender_user == transfer.sender_user]
+
+
+def regular_payment_filter(transfer: CreditTransfer):
+    return [CreditTransfer.transfer_subtype == TransferSubTypeEnum.STANDARD]
+
+
+def matching_transfer_type_filter(transfer: CreditTransfer):
+    return [CreditTransfer.transfer_type == transfer.transfer_type]
+
+
+def matching_transfer_type_and_subtype_filter(transfer: CreditTransfer):
+    return [
+        CreditTransfer.transfer_type == transfer.transfer_type,
+        CreditTransfer.transfer_subtype == transfer.transfer_subtype
+    ]
+
+
+def withdrawal_or_agent_out_and_not_excluded_filter(transfer: CreditTransfer):
+    return [
+        or_(CreditTransfer.transfer_type == TransferTypeEnum.WITHDRAWAL,
+            CreditTransfer.transfer_subtype == TransferSubTypeEnum.AGENT_OUT),
+        CreditTransfer.exclude_from_limit_calcs == False
+    ]
+
+
+def not_rejected_filter():
+    return [CreditTransfer.transfer_status != TransferStatusEnum.REJECTED]
+
+
+class BaseTransferLimit(ABC):
```

Comment:
can you elaborate what you mean? This here is a purely abstract base class already?

Reply:
Ohhh do you mean because `validate_transfer` and `applies_to_transfer` are concrete methods, and you'd like something even those are abstract?

Reply:
I played around with this, but given that the two concrete methods here are pretty opinionated, i think it's cleaner to just let someone whos implementing a new method override those methods if needed, rather than introducing another layer to the class hierarchy just for the sake of being able to put an `abstract` decorator on them.  I've refactored a bit to make the extension process more clear though, with a completely overriden class implemented directly below